stdlib/ballerina-http/src/main/ballerina/net.http.authadaptor/authz-filter.bal

```diff
@@ -0,0 +1,134 @@
+    } else {
+        // if scopes are not defined, no need to authorize
+        return createAuthzResult(true);
+    }
+    return createAuthzResult(authorized);
+}
+
+@Description {value:"Creates an instance of FilterResult"}
+@Param {value:"authorized: authorization status for the request"}
+@Return {value:"FilterResult: Authorization result to indicate if the request can proceed or not"}
+function createAuthzResult (boolean authorized) returns (http:FilterResult) {
+    http:FilterResult requestFilterResult;
+    if (authorized) {
+        requestFilterResult = {canProceed:true, statusCode:200, message:"Successfully authorized"};
+    } else {
+        requestFilterResult = {canProceed:false, statusCode:403, message:"Authorization failure"};
+    }
+    return requestFilterResult;
+}
+
+@Description {value:"Retrieves the scope for the resource, if any"}
+@Param {value:"context: FilterContext object"}
+@Return {value:"string: Scope name if defined, else null"}
+function getScopeForResource (http:FilterContext context) returns (string) {
+    string scope = getAuthzAnnotation(internal:getResourceAnnotations(context.serviceType,
+                                                                      context.resourceName));
+    match scope {
+        string scopeVal => {
+            return scopeVal;
+        }
+        error err => {
+            // if not found in resource level, check in service level
+            scope = getAuthzAnnotation(internal:getServiceAnnotations(context.serviceType));
+            match scope {
+                string scopeVal => {
+                    return scopeVal;
+                }
+                error err => {
+                    // if the scope is still null, means authorization is not needed.
+                    return "";
+                }
+            }
+        }
+    }
+    return "";
+}
+
+@Description {value:"Tries to retrieve the annotation value for scope hierarchically - first from the resource level
+and then from the service level, if its not there in the resource level"}
+@Param {value:"annData: array of annotationData instances"}
+@Return {value:"string: Scope name if defined, else null"}
+function getAuthzAnnotation (internal:annotationData[] annData) returns (string|error) {
+    if (lengthof annData == 0) {
+        error err = {message:"empty annotationData"};
+        return err;
+    }
+    internal:annotationData authAnn = {};
+    foreach ann in annData {
+        if (ann.name == AUTH_ANN_NAME && ann.pkgName == AUTH_ANN_PACKAGE) {
+            authAnn = ann;
+            break;
+        }
+    }
+    if (lengthof authAnn == 0) {
+        // no annotation found for ballerina.auth:config
+        error err = {message:"no annotation for ballerina.auth:Config"};
+        return err;
+    }
+    var authConfig = <auth:AuthConfig> authAnn.value;
+    match authConfig {
+        auth:AuthConfig authConfiguration => {
```

Comment:
You don't need match here. You can use ?=

Reply:
fixed.